### PR TITLE
sim/Make.defs: add -fvisibility=hidden to CFLAGS

### PR DESCRIPTION
**Pull request by GUIDINGLI**
This can hidden all nuttx's symbols,
and DO NOT export nuttx's symbols to share libraries.

e.g.
wrong:

NUTTX       PC       NUTTX
ffmpeg -> asound -> sysconf

right:

NUTTX       PC       PC
ffmpeg -> asound -> sysconf

## Testing
Please ignore the false nxstyle alarm:
```
nuttx/arch/sim/src/sim/up_deviceimage.c:302:18: error: Mixed case identifier found
```

### CODE REVIEW

File: arch/sim/src/nuttx-names.in [hunk 1]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(strncasecmp)
-NXSYMBOLS(strncmp)
 NXSYMBOLS(strncpy)
-NXSYMBOLS(strnlen)
-NXSYMBOLS(strrchr)
-NXSYMBOLS(strstr)
-NXSYMBOLS(strtok_r)
-NXSYMBOLS(strtol)
-NXSYMBOLS(strtoul)
 NXSYMBOLS(syslog)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:100]: should call nuttx's one?

File: arch/sim/src/nuttx-names.in [hunk 2]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(strcpy)
-NXSYMBOLS(strcspn)
-NXSYMBOLS(strdup)
 NXSYMBOLS(strerror)
-NXSYMBOLS(strerror_r)
-NXSYMBOLS(strftime)
-NXSYMBOLS(strlen)
-NXSYMBOLS(strncasecmp)
-NXSYMBOLS(strncmp)
 NXSYMBOLS(strncpy)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:99]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 3]
```diff
@@ -34,190 +34,72 @@
...
 NXSYMBOLS(sleep)
 NXSYMBOLS(snprintf)
 NXSYMBOLS(socket)
-NXSYMBOLS(sprintf)
 NXSYMBOLS(stat)
-NXSYMBOLS(statfs)
 NXSYMBOLS(statvfs)
-NXSYMBOLS(strcat)
-NXSYMBOLS(strchr)
 NXSYMBOLS(strcmp)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:97]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 4]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(sigdelset)
 NXSYMBOLS(sigemptyset)
 NXSYMBOLS(sigfillset)
-NXSYMBOLS(sighold)
-NXSYMBOLS(sigismember)
-NXSYMBOLS(sigprocmask)
-NXSYMBOLS(sigrelse)
-NXSYMBOLS(sigtimedwait)
 NXSYMBOLS(sleep)
 NXSYMBOLS(snprintf)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:93]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 5]
```diff
@@ -34,190 +34,72 @@
...
 NXSYMBOLS(open)
 NXSYMBOLS(opendir)
-NXSYMBOLS(optarg)
-NXSYMBOLS(optind)
-NXSYMBOLS(optopt)
-NXSYMBOLS(pathconf)
 NXSYMBOLS(perror)
-NXSYMBOLS(pipe)
 NXSYMBOLS(poll)
 NXSYMBOLS(printf)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:65]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 6]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(nanosleep)
-NXSYMBOLS(nftw)
-NXSYMBOLS(ntohs)
 NXSYMBOLS(open)
 NXSYMBOLS(opendir)
-NXSYMBOLS(optarg)
-NXSYMBOLS(optind)
-NXSYMBOLS(optopt)
-NXSYMBOLS(pathconf)
 NXSYMBOLS(perror)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:63]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 7]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(ftell)
 NXSYMBOLS(ftruncate)
-NXSYMBOLS(ftw)
-NXSYMBOLS(fwrite)
-NXSYMBOLS(getchar)
-NXSYMBOLS(getenv)
-NXSYMBOLS(getopt)
-NXSYMBOLS(getpid)
-NXSYMBOLS(gettimeofday)
 NXSYMBOLS(htons)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:49]: call nuttx?

File: arch/sim/src/nuttx-names.in [hunk 8]
```diff
@@ -34,190 +34,72 @@
...
 NXSYMBOLS(stat)
-NXSYMBOLS(statfs)
 NXSYMBOLS(statvfs)
-NXSYMBOLS(strcat)
-NXSYMBOLS(strchr)
 NXSYMBOLS(strcmp)
-NXSYMBOLS(strcpy)
-NXSYMBOLS(strcspn)
-NXSYMBOLS(strdup)
 NXSYMBOLS(strerror)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:98]: call nuttx

File: arch/sim/src/nuttx-names.in [hunk 9]
```diff
@@ -34,190 +34,72 @@
 NXSYMBOLS(__cxa_atexit)
 NXSYMBOLS(__errno)
 NXSYMBOLS(_exit)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:36]: remove unused function

File: arch/sim/src/nuttx-names.in [hunk 10]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(fcntl)
-NXSYMBOLS(fdopen)
-NXSYMBOLS(fgetc)
-NXSYMBOLS(fopen)
-NXSYMBOLS(fpathconf)
 NXSYMBOLS(fprintf)
-NXSYMBOLS(fputc)
-NXSYMBOLS(fputs)
-NXSYMBOLS(fread)
 NXSYMBOLS(free)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:45]: call nuttx's

File: arch/sim/src/nuttx-names.in [hunk 11]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(if_indextoname)
 NXSYMBOLS(if_nametoindex)
 NXSYMBOLS(ioctl)
-NXSYMBOLS(isatty)
-NXSYMBOLS(kill)
-NXSYMBOLS(link)
-NXSYMBOLS(listen)
 NXSYMBOLS(lseek)
-NXSYMBOLS(mallinfo)
 NXSYMBOLS(malloc)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:53]: call nuttx's

File: arch/sim/src/nuttx-names.in [hunk 12]
```diff
@@ -34,190 +34,72 @@
...
-NXSYMBOLS(malloc_init)
 NXSYMBOLS(memcmp)
 NXSYMBOLS(memcpy)
 NXSYMBOLS(memset)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:56]: call nuttx's

File: arch/sim/src/nuttx-names.in [hunk 13]
```diff
@@ -34,190 +34,72 @@
...
 NXSYMBOLS(mkdir)
-NXSYMBOLS(mkfifo)
 NXSYMBOLS(mktime)
```
Comment (xiaoxiang781216) [inline arch/sim/src/nuttx-names.in:58]: call nuttx's

**Comment by davids5** [comment]
@GUIDINGLI - just curious can you use --wrap  or weak to accomplish this?

**Comment by xiaoxiang781216** [comment]
> @GUIDINGLI - just curious can you use --wrap or weak to accomplish this?

1. The problem just exist in sim arch, it isn't good to add weak for the common code because the change will impact other arch.
2. It's hard to predict what API the host software will call with the complex simulation(leaverage PC display, mouse, audio, usb etc),  which mean we need add all POSIX/libc/libc++ symbols to the list. This is a huge effort and an unmaintainable approach.
The best method is directly disable host .so call into nuttx elf.

**Pull request merged**